yaks/lib/yaks/default_policy.rb

```diff
@@ -17,13 +19,59 @@ def initialize(options = {})
       @options = DEFAULTS.merge(options)
     end
 
+    # Main point of entry for mapper derivation. Calls
+    # derive_mapper_from_collection or derive_mapper_from_item
+    # depending on the model.
+    #
     # @param model [Object]
     # @return [Class] A mapper, typically a subclass of Yaks::Mapper
     #
-    # @raise [NameError] only occurs when the model is anything but a collection.
+    # @raise [RuntimeError] occurs when no mapper is found or a collection
+    #                       is given but does not respond to :first
     def derive_mapper_from_object(model)
+      mapper = detect_configured_mapper_for(model)
+      return mapper if mapper
       return derive_mapper_from_collection(model) if model.respond_to? :to_ary
-      derive_mapper_from_single_object(model)
+      derive_mapper_from_item(model)
+    end
+
+    # Derives a mapper from the given collection.
+    #
+    # @param collection [Object]
+    # @return [Class] A mapper, typically a subclass of Yaks::Mapper
+    #
+    # @raise [RuntimeError] when collection does not respond to :first
+    def derive_mapper_from_collection(collection)
+      raise_not_a_collection(collection) unless collection.respond_to?(:first)
```

Comment:
ok sure, that kind of usage you can expect, but if they pass garbage in there and get a `NoMethodError` then that's to be expected. That's Ruby's version of a type error.